psl/parser-database/src/types.rs

```diff
@@ -3,11 +3,9 @@ pub(crate) mod index_fields;
 use crate::{context::Context, interner::StringId, walkers::IndexFieldWalker, DatamodelError};
 use either::Either;
 use enumflags2::bitflags;
+use rustc_hash::FxHashMap as HashMap;
 use schema_ast::ast::{self, WithName};
-use std::{
-    collections::{BTreeMap, HashMap},
-    fmt,
-};
+use std::{collections::BTreeMap, fmt};
```

Comment:
What are the specific reasons to opt for a BTreeMap or HashMap in psl? for frequently read, small maps, BTreeMap should have better cache locality at a cost of a slightly slower search.

Reply:
In PSL the decision tree is very simple:

- Do we need range queries? -> BTreeMap
- Else -> HashMap

These maps are global, for the whole schema, usually, so there's only one instance of each per schema and the assumption would be that they tend to grow large (few large data structures is usually better than many small data structures, for heap fragmentation).